src/Microsoft.ML.StandardLearners/Standard/Online/AveragedPerceptron.cs

```diff
@@ -59,9 +59,9 @@ public AveragedPerceptronTrainer(IHostEnvironment env, Arguments args)
 
             _outputColumns = new[]
             {
-                new SchemaShape.Column(DefaultColumnNames.Score, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false),
-                new SchemaShape.Column(DefaultColumnNames.Probability, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false),
-                new SchemaShape.Column(DefaultColumnNames.PredictedLabel, SchemaShape.Column.VectorKind.Scalar, BoolType.Instance, false)
+                new SchemaShape.Column(DefaultColumnNames.Score, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false, new SchemaShape(MetadataForScoreColumn())),
+                new SchemaShape.Column(DefaultColumnNames.Probability, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false, new SchemaShape(MetadataForScoreColumn(true))),
+                new SchemaShape.Column(DefaultColumnNames.PredictedLabel, SchemaShape.Column.VectorKind.Scalar, BoolType.Instance, false, new SchemaShape(MetadataForScoreColumn()))
```

Comment:
>Shape.Column(DefaultColumnNames.PredictedLabel [](start = 26, length = 46)

is it intentional to not use also the input label metadata?

Reply:
Curious: why do you concat to the label metadata of SDCA multiclass, then? To propagate the info about the string - categories conversion? 

---
In reply to: [217769414](https://github.com/dotnet/machinelearning/pull/909#discussion_r217769414) [](ancestors = 217769414,217574282)

Reply:
SDCA propagates key value metadata, that's what I'm doing there. And I think all the multiclass learners do (or should do)

---
In reply to: [217781701](https://github.com/dotnet/machinelearning/pull/909#discussion_r217781701) [](ancestors = 217781701,217769414,217574282)